hera_mc/cm_utils.py

```diff
@@ -126,9 +120,42 @@ def add_date_time_args(parser):
         '--time', help="UTC hh:mm or float (hours), must include --date if use --time", default=0.0)
 
 
+def is_active(at_date, start_date, stop_date):
+    stop_date = get_stopdate(stop_date)
+    return at_date >= start_date and at_date <= stop_date
+
+
+def future_date():
+    """
+    Future is defined here, since defining a far FUTURE_DATE typically gives a
+    warning about UTC vs UT1 etc
+    """
+    return Time.now() + TimeDelta(300, format='jd')
+
+
+def get_stopdate(stop_date):
+    if isinstance(stop_date, Time):
+        return stop_date
+    else:
+        return future_date()
+
+
+def get_displayTime(display):
+    if isinstance(display, str) and display.lower() != 'now':
+        d = display
+    else:
+        d = get_astropytime(display)
+
+    if d is None:
+        d = 'None'
+    elif isinstance(d, Time):
+        d = "{:%Y-%m-%d %H:%M:%S}".format(d.datetime)
+    return d
+
+
 def get_astropytime(_date, _time=0):
     """
-    Take in various incarnations of _date/_time and return an astropy.Time object
+    Take in various incarnations of _date/_time and return an astropy.Time object or None
```

Comment:
I worry about these kinds of swiss-army-knife functions since they make it easy to get sloppy about variable types (namely, the types of whatever variables the caller is passing to this function), which is pretty reliable route to buggy code.

One thing that this function clearly does well is handle textual command line arguments expressing times. That's a nicely focused functionality with a consistent input type. In the other cases that this function handles —  `astropy.Time` objects, integers, None, False — the function is not doing much. Can the non-text input cases be split into a function separate from the one that parses text? At the call sites, we should strive to avoid situations where the types of the date/time variables are ambiguous.

Reply:
Also, the overall clarity of this function could be improved greatly by peeling off cases with the general form

```
if (input is of type X):
   return (appropriate output)
```

Rather than chains of if-else statements and a `return_date` variable. Returning as early as possible prevents rightward drift and helps the reader understand which cases have been completely dealt with and therefore need not be worried about for the rest of the function.